CMakeLists.txt

```diff
@@ -271,17 +253,31 @@ target_include_directories(RAJA
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
   $<INSTALL_INTERFACE:include>)
-target_include_directories(RAJA SYSTEM
-  PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/tpl/cub>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/tpl/rocPRIM/rocprim/include>)
+if (ENABLE_CUDA)
+  if(NOT ENABLE_EXTERNAL_CUB)
+    target_include_directories(RAJA SYSTEM
+      PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/tpl/cub>)
+  endif()
+endif()
+if (ENABLE_HIP)
+  if(NOT ENABLE_EXTERNAL_ROCPRIM)
+    target_include_directories(RAJA SYSTEM
+      PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/tpl/rocPRIM/rocprim/include>)
+  endif()
+endif()
 
 install(DIRECTORY include/ DESTINATION include FILES_MATCHING PATTERN *.hpp)
-if(NOT ENABLE_EXTERNAL_CUB)
-  install(DIRECTORY tpl/cub/ DESTINATION include FILES_MATCHING PATTERN *.cuh)
+if (ENABLE_CUDA)
+  if(NOT ENABLE_EXTERNAL_CUB)
+    install(DIRECTORY tpl/cub/ DESTINATION include FILES_MATCHING PATTERN *.cuh)
+  endif()
 endif()
-if(NOT ENABLE_EXTERNAL_ROCPRIM)
-  install(DIRECTORY tpl/rocPRIM/rocprim/include/ DESTINATION include FILES_MATCHING PATTERN *.hpp)
+if (ENABLE_HIP)
+  if(NOT ENABLE_EXTERNAL_ROCPRIM)
+    install(DIRECTORY tpl/rocPRIM/rocprim/include/ DESTINATION include FILES_MATCHING PATTERN *.hpp)
+  endif()
```

Comment:
I changed this so that non-external cub and rocprim are only installed with RAJA when we are using cuda or hip respectively. This makes sense to when we are using hip to target amd gpus, but if we are using hip to target nvidia gpus we need to install cub instead of rocprim. Do we have a configure option for this so we can easily make this distinction?

Reply:
I don't think so. @davidbeckingsale would it be best to add support for this to BLT?

Reply:
Currently the configuration with hip targeting cuda is broken. Shall I always look for cub or install cub when hip is enabled (as was done previously) to workaround the issue until we have enough information to do this right?

Reply:
Whatever it takes to get it working for now, I guess. Do we have to worry about version issues between rocm and cub?

Reply:
Not that I know of. In this case you are really using nvcc so the main version issues will be the same one this PR is trying to address.
I'm thinking of adding ENABLE_CUB that defaults to On when ENABLE_CUDA or ENABLE_HIP is on and Off otherwise. Do you think this would create confusion because it is always required for cuda builds and hip(cuda) builds?

Reply:
Not sure. @ajkunen @trws @davidbeckingsale what do you guys think?